### ISSUE TITLE
Use of is_unbalance parameter

### ISSUE THREAD

**Issue by jsanjay63** [microsoft/LightGBM#4514]
## Description
I am trying to use is_unbalance parameter in my model training for binary classification problem where positive class is approx 3%. If I set the parameter 'is_unbalance', I observe that the binary log loss drops in the first iteration but then keeps on increasing. I'm noticing this behaviour only if I enable this parameter 'is_unbalance'. Appreciate your help on this. Thanks

## Reproducible example
{'learning_rate': 0.03,  'is_unbalance': True, 'objective': 'binary', 'verbosity': 2}

## Environment info

LightGBM version or commit hash: 3.2.1

## Additional Comments

![download](https://user-images.githubusercontent.com/19145015/128992371-32101e24-f726-4ab4-bda0-eaea230ec74a.png)

**Comment by jameslamb** [comment]
Thanks very much for using LightGBM.

There is not enough information in your report for maintainers here to assist you without significant guessing. For example, you haven't mentioned the programming language you're using or the operating system you're on.

Are you able to provide a reproducible example, some self-contained code we could run to confirm this behavior?
